### ISSUE TITLE
Campaign for a segment with 500+ leads does not run all events (only the first)

### ISSUE THREAD

**Issue by adonaicruz** [mautic/mautic#6419]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | YES
| Feature request? | NO
| Enhancement? | NO

## Description:
I've created a campaign for a **500+ lead segment**. But the **first** 'campaign: trigger' does not execute all events. 
When a single lead enters the campaign after the first trigger, events execute normally.
I send below some images to exemplify:

obs.: I did a test on a new mautic installation

![Print1](http://amazon.tekoa.floripa.br/download/prints/mautic1.jpg "Print1")
![Print2](http://amazon.tekoa.floripa.br/download/prints/mautic2.jpg "Print2")
![Print3](http://amazon.tekoa.floripa.br/download/prints/mautic3.jpg "Print3")

```
triggering event for campaign 21
Triggering Events for Recent Recent Contacts
% events% total event (s) to be processed in batches of 100 contacts
  561/561 [============================] 100%

2 events in total were executed
A total of {0} 0 events were scheduled

Shooting Scheduled Events
% events% total event (s) to be processed in batches of 100 contacts
A total of {0} 0 events were executed
A total of {0} 0 events were scheduled
```

## If a bug:

| Q   | A
| --- | ---
| Mautic version | v2.14.0
| PHP version | 7.1.17

### Steps to reproduce:
1. Create a segment with 500 leads
2. Create a campaign with this segment
3. Create a simple flow with some events:
  3.1. Action 1: Add tag 'success';
  3.2. Action 2: restart campaign (1 day delay);
 
### Log errors: 

no related errors were generated in [mautic root]/app/log/ or [apache|php]/log/error_log

**Comment by adonaicruz** [comment]
i find a way to fix this problem
```
php app/console mautic:campaigns:trigger --max-threads=3 --batch-limit=6
```
but if you keep the default settings the problem persists.
I have reproduced in a clean installation of mautic, so it is not migration problem. Just create a simple campaign and a event (to add a tag for example)

**Comment by heathdutton** [comment]
Hello @adonaicruz.  The `--max-threads` parameter must be used in conjunction with the `--thread-id` parameter, otherwise it will not do anything.

The first trigger run should only process the "kickoff" events and will stop for the next cron job to pick up the next batch, this is normal. How is the "Is Today" condition set up (as I suspect your issue is there)? 

Perhaps I am not understanding the issue.

**Comment by adonaicruz** [comment]
@heathdutton  Thank you for your help.
I believe it is not the same problem  of  #6211. I did a test with this fix and the bug continues and Nothing happens in the next cron.
I tested with a simpler campaign like this below, with no results:
![mautic4](https://user-images.githubusercontent.com/5114795/43841350-53fe43aa-9af9-11e8-90a4-cbc4ba40bbe0.jpg)

**Comment by levrykurniawan** [comment]
It seems like there is an issue on the campaign trigger. I also found similar issue when running a campaign. I think #6433 is related to this too. The issue will disappear if set a small value on batch limit.

**Comment by dsp76** [comment]
Seems to be the same issue I ran into when trying to validate emails of an existing segment? 
The segment contains 626 contacts and only runs for the first couple of users.

![screenshot_2018-08-26 kampagne andern mautic](https://user-images.githubusercontent.com/1241376/44645819-4aa31800-a9d9-11e8-84c0-8085a68c71ee.png)

**Comment by heathdutton** [comment]
Thanks all, I believe the issue in question will be solved by #6482

Reopen if the issue persists after 2.14.1
